src/SamplesApp/SamplesApp.UITests/UnoSamples_Tests.Keyboard.cs

```diff
@@ -285,6 +285,54 @@ public void TextBox_PageLoadedTest()
 			_app.WaitForElement(appendOutput);
 		}
 
+		[Test]
+		public void TextBox_TextChanging_Capitalize()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.TextBoxTests.TextBox_TextChanging");
+
+			var textBlock = _app.Marked("CapitalizePreviousTextBlock");
+
+			const string Entered = "This patience is a virtue";
+			const string Final = "THIS PATIENCE IS A VIRTue";
+
+			var textBox = TypeInto("CapitalizePreviousTextBox", Entered, Final);
+
+			Assert.AreEqual(Final, GetText(textBox));
+			Assert.AreEqual(Final, GetText(textBlock));
+		}
+
+		[Test]
+		public void TextBox_TextChanging_Limit()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.TextBoxTests.TextBox_TextChanging");
+
+			var textBlock = _app.Marked("LimitLengthTextBlock");
+
+			const string Entered = "abcdefghijklmnopqr";
+			const string Final = "defghijklmnopqr";
+
+			var textBox = TypeInto("LimitLengthTextBox", Entered, Final);
+
+			Assert.AreEqual(Final, GetText(textBox));
+			Assert.AreEqual(Final, GetText(textBlock));
+		}
+
+		[Test]
+		public void TextBox_BeforeTextChanging()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.TextBoxTests.TextBox_BeforeTextChanging");
+
+			var textBlock = _app.Marked("BeforeTextBlock");
+
+			const string Entered = "reassesses";
+			const string Final = "eee";
+
+			var textBox = TypeInto("BeforeTextBox", Entered, Final);
+
+			Assert.AreEqual(Final, GetText(textBox));
+			Assert.AreEqual(Final, GetText(textBlock));
+		}
```

Comment:
Could you add a test that validates that clearing and entering again works as well ?

Reply:
Gotcha, will do